subsys/bluetooth/audio/csis_client.c

```diff
@@ -1285,6 +1310,37 @@ static int verify_members_and_get_inst(const struct bt_csis_client_set_member **
 	return 0;
 }
 
+int bt_csis_client_get_lock_state(const struct bt_csis_client_set_member **members,
+				   uint8_t count,
+				   const struct bt_csis_client_set *set)
+{
+	int err;
+
+	if (busy) {
+		BT_DBG("csis_client busy");
+		return -EBUSY;
+	}
+
+	cur_inst = NULL;
+	err = verify_members_and_get_inst(members, count, set, true, &cur_inst);
+	if (err != 0) {
+		BT_DBG("Could not verify members: %d", err);
+		return err;
+	}
+
+	err = csis_client_read_set_lock(cur_inst);
```

Comment:
do we need to read it always? Don't we register for the notifications and maintain the current lock state?

So in fact, we need to read it once, after bonding, and then we should rely on notifications if we are bonded.
If not bonded, that read it only once after reconnection, and then rely on the notifications.

Reply:
Notifications on ATT are not reliable.  It seems to follow that one should not rely upon them ;-)

Reply:
> Notifications on ATT are not reliable. It seems to follow that one should not rely upon them ;-)

**unreliable** is really the keyword to consider when determining if you should **rely** on it indeed :D